sentry/sentry.go

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package sentry
+
+import (
+	"time"
+
+	raven "github.com/getsentry/raven-go"
+	"github.com/pkg/errors"
+	"github.com/uber-go/zap"
+	"github.com/uber-go/zap/zwrap"
+)
+
+var (
+	_platform          = "go"
+	_traceContextLines = 3
+	_traceSkipFrames   = 3
+	_zapToRavenMap     = map[zap.Level]raven.Severity{
+		zap.DebugLevel: raven.DEBUG,
+		zap.InfoLevel:  raven.INFO,
+		zap.WarnLevel:  raven.WARNING,
+		zap.ErrorLevel: raven.ERROR,
+		zap.PanicLevel: raven.ERROR,
+		zap.FatalLevel: raven.FATAL,
+	}
+)
+
+// Logger automatically sends logs above a certain threshold to Sentry
+type Logger struct {
+	zap.Meta
+	Capturer
+
+	// Minimum level threshold for sending a Sentry event
+	minLevel zap.Level
+
+	// Cotrols if stack trace should be automatically generated, and how many frames to skip
+	traceEnabled      bool
+	traceSkipFrames   int
+	traceContextLines int
+	traceAppPrefixes  []string
+
+	// Additional information that should be attached to each packet, i.e. customerUUID
+	extra zwrap.KeyValueMap
+}
+
+// Option pattern for logger creation
+type Option func(l *Logger)
+
+// New Sentry logger
+func New(dsn string, options ...Option) (*Logger, error) {
+	l := &Logger{
+		minLevel:          zap.ErrorLevel,
+		traceEnabled:      true,
+		traceSkipFrames:   _traceSkipFrames,
+		traceContextLines: _traceContextLines,
+		extra:             zwrap.KeyValueMap{},
+	}
+
+	for _, option := range options {
+		option(l)
+	}
+
+	// Default to live sentry packet capturer
+	if l.Capturer == nil {
```

Comment:
@akshayjshah I'm not for a moment suggesting that there will be two versions of a binary running around, I was simply talking about glide and dependency management.

This change would mean that people using `zap/sentry` would also have to import `raven-go` directly in order to use the sentry logger. That means they *could* lock in the version of raven-go in their glide.yaml, which then can lead to trouble. We run into this situation pretty frequently with uberfx, since we're acting as a middle man for a lot of other libraries.

If both of you don't think that it is going to be a usability issue, I have no problem with making the change otherwise.